src/collision/cm_trace.c

```diff
@@ -24,7 +24,7 @@
 /**
  * @brief Box trace data encapsulation and context management.
  */
-static __thread struct {
+typedef struct {
```

Comment:
I've read in many places that TLS is slow -- probably slower than passing an additional parameter down the call chain. I also feel that, stylistically, param passing is cleaner. Was there a compelling reason to use TLS? I changed this as a separate commit, so it's easy to undo if necessary.